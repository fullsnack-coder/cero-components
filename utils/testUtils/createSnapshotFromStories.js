import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
