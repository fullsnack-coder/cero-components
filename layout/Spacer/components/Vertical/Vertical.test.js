import * as stories from './Vertical.stories'
import createSnapshotFromStories from '../../../../utils/testUtils/createSnapshotFromStories'

describe('[ layout / Vertical ]', () => {
  describe('#Vertical', () => createSnapshotFromStories('Vertical', stories))
})
