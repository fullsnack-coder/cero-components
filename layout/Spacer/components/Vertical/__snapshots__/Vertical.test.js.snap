// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ layout / Vertical ] #Vertical when 'Vertical' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer vertical is-playground"
    style="height: 0px;"
  />
</DocumentFragment>
`;

exports[`[ layout / Vertical ] #Vertical when 'Vertical' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="spacer vertical is-playground"
    style="height: 0px;"
  />
  <div
    class="spacer vertical is-playground"
    style="height: 5px;"
  />
  <div
    class="spacer vertical is-playground"
    style="height: 15px;"
  />
  <div
    class="spacer vertical is-playground"
    style="height: 20px;"
  />
  <div
    class="spacer vertical is-playground"
    style="height: 40px;"
  />
</DocumentFragment>
`;
